plugin/plugin.go

```diff
@@ -0,0 +1,209 @@
+func validateSeverity(severity string) error {
+	switch severity {
+	case SeverityCritical, SeverityError, SeverityWarning, SeverityInfo, SeverityUnknown:
+		return nil
+	default:
+		return errors.New("invalid severity value; allowed values are 'critical', 'error', 'warning', 'info', 'unknown'")
+	}
+}
+
+// Exec executes the plugin.
+func Exec(ctx context.Context, client PagerDutyClient, args Args) error {
+	logger := logrus.WithFields(logrus.Fields{
+		"PLUGIN_ROUTING_KEY":         args.RoutingKey,
+		"PLUGIN_INCIDENT_SUMMARY":    args.IncidentSummary,
+		"PLUGIN_INCIDENT_SOURCE":     args.IncidentSource,
+		"PLUGIN_INCIDENT_SEVERITY":   args.IncidentSeverity,
+		"PLUGIN_CREATE_CHANGE_EVENT": args.CreateChangeEvent,
+		"PLUGIN_JOB_STATUS":          args.JobStatus,
+	})
+
+	logger.Info("Starting plugin execution")
+
+	// Validate required fields
+	if args.RoutingKey == "" {
+		return errors.New("missing required parameter: routingKey")
+	}
+	if args.IncidentSummary == "" {
+		return errors.New("missing required parameter: incidentSummary")
+	}
+	if args.IncidentSource == "" {
+		return errors.New("missing required parameter: incidentSource")
+	}
+	if !args.CreateChangeEvent {
+		if args.DedupKey == "" {
+			return errors.New("missing required parameter: dedupKey when not creating a change event")
+		}
+		if args.JobStatus == "" {
+			return errors.New("missing required parameter: jobStatus when not creating a change event")
+		}
+	}
+
+	// Validate severity value if not creating a change event
+	if !args.CreateChangeEvent {
+		if err := validateSeverity(args.IncidentSeverity); err != nil {
+			return err
+		}
+	}
+
+	if args.JobStatus == "" {
+		logger.Warn("Job status is empty, no action will be taken")
+	}
+
+	if args.CreateChangeEvent {
+		logger.Info("Creating change event")
+		if err := createChangeEvent(ctx, client, args); err != nil {
+			logger.WithError(err).Error("Failed to create change event")
+			return errors.New("failed to create change event")
+		}
+		logger.Info("Change event created Successfully")
+		return nil
+	}
+
+	// Handle job status and decide whether to trigger or resolve incidents
+	var resolveIncident bool
+	var summary = args.IncidentSummary
+
+	switch args.JobStatus {
+	case "success":
+		resolveIncident = args.Resolve
+		summary = "Job succeeded: " + summary
+		logger.Info("Job succeeded, deciding on resolving incident")
+	case "failure":
+		resolveIncident = false
+		summary = "Job failed: " + summary
+		logger.Info("Job failed, deciding on triggering or resolving incident")
+	case "unstable":
+		resolveIncident = false
+		summary = "Job is unstable: " + summary
+		logger.Info("Job is unstable, deciding on triggering or resolving incident")
+	case "aborted":
+		resolveIncident = false
+		summary = "Job was aborted: " + summary
+		logger.Info("Job was aborted, deciding on triggering or resolving incident")
+	default:
+		logger.Warn("Unknown job status, no action taken")
+		return nil
```

Comment:
do you think we should use `SeverityUnknown` for undefined job statuses?
```suggestion
	default:
	        summary = "Job status unknown: " + summary
		logger.Warn("Unknown job status, no action taken")
		return nil
```